### ISSUE TITLE
if terminal height is small, on `s` press it panics in ratatui-0.26.0/src/buffer/buffer.rs:108

### ISSUE THREAD

**Issue by wis** [Xithrius/twitch-tui#551]
**Describe the bug:**

if the terminal height is small enough, maybe smaller than the menu that appears when you press `s`,  
it panics with the following crash message:

```
The application panicked (crashed).
Message:  index out of bounds: the len is 2352 but the index is 2372
Location: /home/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ratatui-0.26.0/src/buffer/buffer.rs:108

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
channel closed
```

**To Reproduce:**

- go to home screen of twt
- reresize terminal to be shorter than it is wide, by a factor of 2 or 3.
- press <kbd>`s`</kbd>

**Expected behavior:**

1st it definitely should not crash, 2nd maybe it should render a smaller menu if the terminal height is small 

**Screenshots:**

not needed, crash message is pasted above.

**System:**

- Twitch-tui version:  `twitch-tui 2.6.3`
  - installed from AUR with `yay -S twitch-tui` 

- Operating System: Arch Linux on WSL 
`Linux myhostname 5.15.133.1-microsoft-standard-WSL2 #1 SMP Thu Oct 5 21:02:42 UTC 2023 x86_64 GNU/Linux`

- Terminal: Windows Terminal

- Shell: `zsh 5.9 (x86_64-pc-linux-gnu)`

**Additional context:**
I run twt in Tmux because I run everything in Tmux.

**Comment by Xithrius** [comment]
Thanks for the issue, I'll take a stab at this hopefully later today.

**Comment by Xithrius** [comment]
Ah, this was fixed in 2.6.4 (able to reproduce on 2.6.3, and most recent version is 2.6.5), but I see the AUR is on 2.6.3.

Once you upgrade, all should be good.

**Comment by Xithrius** [comment]
Looks like the AUR is now up to date. If you still have this issue, feel free to reopen.
